lib/logflare/backends.ex

```diff
@@ -478,21 +479,62 @@ defmodule Logflare.Backends do
     end
   end
 
+  def push_recent_events(%Source{}, []), do: :ok
+
+  def push_recent_events(%Source{id: source_id}, log_events) do
+    existing = :ets.lookup(:recent_events, source_id)
+
+    :ets.delete(:recent_events, source_id)
```

Comment:
At least use `:ets.take` here yeah?

Reply:
Ah yes we can do that since we're already limiting count on insert